spec/java/lang/Double.main.lsl

```diff
@@ -0,0 +1,375 @@
+libsl "1.1.0";
+
+library std
+    version "11"
+    language "Java"
+    url "https://github.com/openjdk/jdk11/blob/master/src/java.base/share/classes/java/lang/Double.java";
+
+// imports
+
+import java/lang/Object;
+import java/lang/String;
+
+import java/lang/Double;
+
+
+// automata
+
+automaton DoubleAutomaton
+(
+    var value: double = 0.0,
+)
+: LSLDouble
+{
+    // states and shifts
+
+    initstate Allocated;
+    state Initialized;
+
+    shift Allocated -> Initialized by [
+        // constructors
+        Double (LSLDouble, String),
+        Double (LSLDouble, double),
+
+        // static operations
+        compare,
+        doubleToLongBits,
+        doubleToRawLongBits,
+        hashCode (double),
+        isFinite,
+        isInfinite (double),
+        isNaN (double),
+        longBitsToDouble,
+        max,
+        min,
+        parseDouble,
+        sum,
+        toHexString,
+        toString (double),
+        valueOf (String),
+        valueOf (double),
+    ];
+
+    shift Initialized -> self by [
+        // instance methods
+        byteValue,
+        compareTo,
+        doubleValue,
+        equals,
+        floatValue,
+        hashCode (LSLDouble),
+        intValue,
+        isInfinite (LSLDouble),
+        isNaN (LSLDouble),
+        longValue,
+        shortValue,
+        toString (LSLDouble),
+    ];
+
+    // internal variables
+
+    // utilities
+
+    @static proc _getRawBits (v: double): long
+    {
+        if (v != v) // a NaN?
+            result = 9221120237041090560L;
+        else if (1.0 / v == NEGATIVE_INFINITY) // is it a "-0.0" ?
```

Comment:
This is the only way I can think of to make it possible to check for *negative zeros* since we do not have direct bit-level access to underlying structures of `float`s and `double`s.

Reply:
@dog-m Why we can't write such checking: "v == -0.0" ? Or LibSL can't work with such construction ?